internal/nix/nixprofile/profile.go

```diff
@@ -135,12 +135,13 @@ func ProfileListIndex(args *ProfileListIndexArgs) (int, error) {
 		if err != nil {
 			return -1, errors.Wrapf(err, "failed to get installable for %s", args.Package.String())
 		}
-		for _, item := range items {
-			if ref == item.unlockedReference {
-				return item.index, nil
+		if ref != "" { // ref is empty when package is flake output (and not a devbox package)
+			for _, item := range items {
+				if ref == item.unlockedReference {
+					return item.index, nil
+				}
```

Comment:
I'd rather implement this as:
```
if !inCache && args.Package.IsDevboxPackage() {
 
   ...
   return -1, errors.Wrap(nix.ErrPackageNotFound, args.Package.String())
}
```
so that the if-statement does not fallthrough and we don't have to consider DevboxPackages when reasoning about the code below.

Alternatively, is this optimization saving meaningful time? If not, lets drop it.